ThirdPartyAdapters/chartboost/chartboost/src/main/java/com/google/ads/mediation/chartboost/ChartboostAdapterUtils.java

```diff
@@ -174,4 +179,48 @@ static int getAdRequestErrorType(CBError.CBImpressionError error) {
                 return AdRequest.ERROR_CODE_NO_FILL;
         }
     }
+
+    /**
+     * Parse and report Chartboost banner error code to AdMob error
+     * @param code
+     */
+    static int parseChartboostErrorCodeToAdMobErrorCode(int code) {
+        switch (code) {
+            case 1:
+            case 5:
+            case 7:
+                return AdRequest.ERROR_CODE_NETWORK_ERROR;
+            case 6:
+                return AdRequest.ERROR_CODE_NO_FILL;
+            case 16:
+                return AdRequest.ERROR_CODE_INVALID_REQUEST;
+            default:
+                return AdRequest.ERROR_CODE_INTERNAL_ERROR;
+        }
+    }
+
+    /**
+     * Calculate possible Chartboost banner size format based on the provided AdSize
+     * @param adSize
+     * @return
+     */
+    static BannerSize chartboostAdSizeFromLocalExtras(AdSize adSize) {
+        if (adSize != null) {
+            int adHeight = adSize.getHeight();
+            int adWidth = adSize.getWidth();
+            int LEADERBOARD_HEIGHT = BannerSize.getHeight(BannerSize.LEADERBOARD);
+            int LEADERBOARD_WIDTH = BannerSize.getWidth(BannerSize.LEADERBOARD);
+            int MEDIUM_HEIGHT = BannerSize.getHeight(BannerSize.MEDIUM);
+            int MEDIUM_WIDTH = BannerSize.getWidth(BannerSize.MEDIUM);
+
+            if (adHeight >= LEADERBOARD_HEIGHT && adWidth >= LEADERBOARD_WIDTH) {
+                return BannerSize.MEDIUM;
+            } else if (adHeight >= MEDIUM_HEIGHT && adWidth >= MEDIUM_WIDTH) {
+                return BannerSize.LEADERBOARD;
+            } else {
+                return BannerSize.STANDARD;
+            }
```

Comment:
This is something that we don't do on iOS, assuming that the server will do the necessary transformations if needed. It would make sense for Android to follow the same pattern. We probably should also confirm with AdServer that they are prepared to do this for Android banners.

Reply:
Also, I think the logic is off. If size is bigger than leaderboard, then use medium?

Reply:
Fixed